app/controllers/posts_controller.rb

```diff
@@ -0,0 +1,42 @@
+class PostsController < ApplicationController
+  def index
+    @user = User.find(params[:user_id])
+  end
+
+  def show
+    @user = User.find(params[:user_id])
+    @post = Post.find(params[:id])
+    @like_user = User.first
+  end
+
+  def create
+    @post = Post.new(post_params)
+    @post.author_id = current_user.id
+    @post.likes_counter = 0
+    @post.comments_counter = 0
+    return unless @post.save
+
+    redirect_to user_posts_path
+  end
```

Comment:
- [ ] @Fahim-Khan-P Greate work on your create method, however, it's good practice to have a redirect route for both success and failure when creating a post, currently, it's only for when a post is created.

Reply:
Thank you for your feedback! I agree that it's good practice to have a redirect route for both success and failure when creating a post. I have updated the create method in the PostsController to handle both cases.